### ISSUE TITLE
Add an option to specify lua binary path/name

### ISSUE THREAD

**Issue by rbgarga** [awesomeWM/awesome#1377]
Output of `awesome --version`:

Impossible to run since it's a build issue

**How to reproduce the issue:**

Just try to build awesome 4.0 on FreeBSD

**Actual result:**

It won't find 'lua' binary because on FreeBSD you can have multiple lua versions installed simultaneously and binaries are called lua51, lua52, lua53, ...

**Expected result:**

Any way to define lua binary path to be used

**Comment by psychon** [comment]
I'll propose this once and then go hide in shame immediately again...
```sh
(echo '#!/bin/sh' ; echo 'lua53 "$*"') > /usr/local/bin/lua
chmod a+x /usr/local/bin/lua
```
From a quick look, affected files are: `tests/examples/CMakeLists.txt`, `build-utils/lgi-check.sh`, `awesomeConfig.cmake`, `tests/_client.lua`

The CMake files could just use a CMake variable (which is a bit harder since @Elv13 wants `tests/examples/CMakeLists.txt` to be callable on its own). Since `lgi-check.sh` is just called by CMake, it could use an environment variable to communicate the expected binary. `_client.lua` is a problem.

Ideas?

Fun fact: CMake's `FindLua` does not find a Lua binary to use.

**Comment by blueyed** [comment]
This also affects the awesome-luajit-git AUR package, which currently patches files to use `luajit` instead of `lua` (https://aur.archlinux.org/cgit/aur.git/commit/PKGBUILD?h=awesome-luajit-git&id=1fc14a9282d821d8fbb25363b5521fd9f664576f).
I thought about changing it to prepend a `lua` wrapper/bin into `$PATH` during build, but that did not seem to help - maybe PATH gets sanitized somewhere?!  (tried prepending it with `env` to `cmake` and `make`).

**Comment by sclu1034** [comment]
This was implemented in #3279. I guess GitHub didn't know to auto-close it.
